.eslintrc

```diff
@@ -1,11 +1,33 @@
 {
-  "plugins": ["react", "jsx-a11y", "prettier"],
+  "env": {
+    "browser": true,
+    "es6": true,
+    "node": true
+  },
+  "plugins": ["react", "@typescript-eslint", "react-hooks", "jsx-a11y"],
   "extends": [
-    "airbnb",
-    "prettier",
-    "prettier/react",
     "plugin:react/recommended",
-    "plugin:jsx-a11y/recommended"
+    "plugin:@typescript-eslint/recommended",
+    "plugin:jsx-a11y/recommended",
+    "airbnb/hooks",
+    "prettier"
+  ],
+  "parser": "@typescript-eslint/parser",
+  "parserOptions": {
+    "ecmaFeatures": {
```

Comment:
Do you need to add `"project": "tsconfig.json",` here or is this something that's added later on once TS is "turned on"?

Reply:
Good questions. I don't believe this is necessary in this case however I'm not certain. Let me investigate.

Reply:
Ok, no. It isn't strictly required. Though it does not impede linting if omitted. 

I'm confident that the purpose of the `project` setting in `parserOptions` is to use the configurations set in a `tsconfig.json` file for files other than your standard `.js, .jsx, .tsx` files.

The below excerpt is supplemental to the `project` configuration in [TypeScript-parser](https://github.com/typescript-eslint/typescript-eslint/tree/main/packages/parser#parseroptionsproject) docs. However, it provides support around the inclusion of which files are to be linted vs ignored.

Regardless though, the relevance or necessity of **NOT** using the `project` configuration is not apparent and I think adding it would be a positive approach to utilizing future type information. Assuming we convert our JSX into TSX.

> parserOptions.ecmaFeatures.jsx
>
> Default false.
>
> Enable parsing JSX when true. More details can be found here.
> NOTE: this setting does not affect known file types (.js, .jsx, .ts, .tsx, .json) because the TypeScript compiler has its own internal handling for known file extensions. The exact behavior is as follows:
> - if parserOptions.project is not provided:
>   - .js, .jsx, .tsx files are parsed as if this is true. 
>   - unknown extensions (.md, .vue) will respect this setting.
> - if parserOptions.project is provided (i.e. you are using rules with type information):
>    - .js, .jsx, .tsx files are parsed as if this is true.
>    - .ts files are parsed as if this is false.
>    - "unknown" extensions (.md, .vue) are parsed as if this is false.